src/sidebar/components/test/util.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Wait for a condition to evaluate to a truthy value.
+ *
+ * @param {() => any} condition - Function that returns a truthy value when some condition is met
+ * @param {number} timeout - Max delay in milliseconds to wait
+ * @param {string} what - Description of condition that is being waited for
+ * @return {Promise<any>} - Result of the `condition` function
+ */
+export async function waitFor(
+  condition,
+  timeout = 10,
+  what = condition.toString()
+) {
+  const result = condition();
+  if (result) {
+    return result;
+  }
+
+  const start = Date.now();
+
+  return new Promise((resolve, reject) => {
+    const timer = setInterval(() => {
```

Comment:
Looks like there's a default of `10`?

Reply:
We probably should change this in LMS too, but it loos like the `milliseconds` param is "required" but it may default to 10 otherwise?

https://www.w3schools.com/jsref/met_win_setinterval.asp

Can you add  ",10" on (I think line 32)

Reply:
Oh, sorry, you were asking a different question than I realized. No, there is no interval delay set here, it is true. My hunch is that @robertknight intended "next tick", so I'd like to talk to him first before adding a `delay`. I can do this as soon as tomorrow.

Reply:
I would generally advise looking at [MDN](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setInterval) or better yet, the relevant web platform specifications which are linked at the bottom of the MDN article, for up-to-date information on web platform APIs.

As far as I can determine from the MDN article, and the linked platform specs, the minimum interval value is 0 and this is the default if not specified. I used to work with native Win32 APIs and I vaguely recall that they had a minimum delay of 10ms for timers, so it is possible that 10 is indeed the minimum in some older browsers.

When using `setTimeout` and `setInterval` in tests, I would generally recommend the shortest duration you can get away with, since that makes the test run faster. In general, avoid writing tests that rely on something happening in a fixed amount of time though since that can cause flakey tests if the build machine is under stress.